ivy/utils/backend/handler.py

```diff
@@ -252,10 +252,11 @@ def _set_backend_as_ivy(
     backend_str = backend.current_backend_str() if backend_str is None else backend_str
     for k, v in original_dict.items():
         compositional = k not in backend.__dict__
-        if k not in backend.__dict__:
+        if compositional:
             if k in invalid_dtypes and k in target.__dict__:
                 del target.__dict__[k]
                 continue
+            v = _verify_efficient_implementations(v) if v else v
```

Comment:
actually we should also be checking for efficient backend implementations for non-compositional functions. 
Ofc as far as `scaled_dot_product_attention` goes there's a compositional implementation and a `xformers` sub-backend implementation with no `torch` backend implementation, but it could be there for some other sub-backend added in the future

Reply:
sure, I'll do that